ads/yahoonativeads.js

```diff
@@ -21,26 +21,30 @@ import {loadScript, validateData} from '../3p/3p';
  * @param {!Object} data
  */
 export function yahoonativeads(global, data) {
-  validateData(data, ['section'], ['module']);
+  validateData(data, ['key', 'code', 'url']);
 
-  (global.readmo = global.readmo || []).push({
-    section: data.section,
-    container: '#c',
-    module: data.module,
-    amp: true,
-  });
+  (global.native = global.native || []).push(data.code);
 
-  global.context.observeIntersection(function(entries) {
-    entries.forEach(function(entry) {
-      if (global.Readmo) {
-        global.Readmo.onViewChange({
-          rects: entry,
-        });
-      }
-    });
-  });
+  global.apiKey = data.key;
+  global.publisherUrl = data.url;
```

Comment:
I can't see your JS but one thing we generally recommend is to make sure you URL encode these values before trying to send requests.

Reply:
To be clear - I mean doing this in your external JS if it's not already there.

Reply:
yes, of course

Reply:
glad you're calling that out in PRs 👍